### DISCUSSION TITLE
use pr instead of directly pushing to master

### DISCUSSION THREAD

**Discussion by alex19EP** [espeak-ng/espeak-ng#1551] [Announcements]
hello. this applies to the main team, which has access to push to the master. since we now have a fairly extensive CI with tests, it would be great if all merges into the master branch were done by creating a prs, this will improve visibility and tractability of changes.

**Comment by jaacoppi** [comment]
I'm guilty!

Good idea. Can GitHub Actions force this behavior?

**Reply by jaacoppi** [reply]
https://stackoverflow.com/questions/46146491/prevent-pushing-to-master-on-github

Some ideas to get started.

**Comment by djphoenix** [comment]
Indeed good idea.
I’m also propose to perform only fast-forward merges, because some fixes may break the code even if its have no direct conflicts. So rebase-test-merge is preferred way for me (and many other projects). Any objections?

**Comment by alex19EP** [comment]
> I’m also propose to perform only fast-forward merges, because some fixes may break the code even if its have no direct conflicts.

implemented.

@jaacoppi I have now turned on the disallow to override restrictions for administrators option. so I think next time you will see an error when pushing directly to the master.
